File: C++/BakedGood.cpp
//
// Created by NandM2 on 12/10/2021.
//

#include "BakedGood.h"

BakedGood::BakedGood(double basePrice) : basePrice(basePrice) {}

double BakedGood::getBasePrice() const {
    return basePrice;
}

std::string BakedGood::ToString() const {
    std::string price = std::to_string(basePrice);
    return "($" + price.substr(0,price.find('.')+3) + ")";
}

File: Python3/BakedGood.py
from abc import ABC, abstractmethod

class BakedGood():
    def __init__(self, basePrice):
        self._basePrice = basePrice

    def __str__(self):
        return "(${:.2f})".format(self._basePrice)

    def getBasePrice(self):
        return self._basePrice

    @abstractmethod
    def DiscountedPrice(self, quantity):
        pass
